src/inputs/dnstap/DnstapInputStream.cpp

```diff
@@ -60,6 +60,19 @@ void DnstapInputStream::_read_frame_stream_file()
                 _logger->warn("dnstap data is wrong type or has no message, skipping frame of size {}", len_data);
                 continue;
             }
+            if (_f_enabled[Filters::OnlyHosts]) {
```

Comment:
let's break this code out into a method which excepts a const ref to the message and returns a bool, and use it in file/udp/tcp instead of repeating these blocks.

Reply:
I think it is not good to throw things in critical section. what about only returning true or false?